lego/apps/companies/views.py

```diff
@@ -126,3 +141,106 @@ def get_serializer_class(self):
         elif self.action in ["create", "update", "partial_update"]:
             return CompanyInterestCreateAndUpdateSerializer
         return CompanyInterestSerializer
+
+    @action(detail=False, methods=["GET"])
+    def csv(self, *args, **kwargs):
+        user = self.request.user
+        is_admin = user.has_perm(EDIT, obj=Company)
+        if not is_admin:
+            return Response(status=status.HTTP_403_FORBIDDEN)
+
+        year = self.request.query_params.get("year")
+        semester_season = self.request.query_params.get("semester")
+        event = self.request.query_params.get("event")
+
+        try:
+            semester = Semester.objects.get(year=year, semester=semester_season)
+        except ObjectDoesNotExist:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+
+        companyInterests = CompanyInterest.objects.filter(semesters__in=[semester])
+        if event:
+            companyInterests = companyInterests.filter(events__contains=[event])
+
+        event_string = f"-{event}" if event else ""
+        response = HttpResponse(content_type="text/csv")
+        response[
+            "Content-Disposition"
+        ] = f'attachment; filename="{f"Company-interests-{year}-{semester}{event_string}"}.csv"'
+
+        writer = csv.writer(response)
+        writer.writerow(
+            [
+                "Navn på bedrift:",
+                "Kontaktperson",
+                "Mail:",
+                "Telefonnumm:",
+                "Bedriftsinformasjon:",
+                "Semester:",
+                "Arrangementer:",
+                "Annet:",
+                "Samarbeid:",
+                "Bedriftstype:",
+                "Relevante temaer:",
+                "Kontorer i Trondheim for besøk:",
+                "Klassetrinn",
+                "Antall deltagere",
+                "Faglig arrangement kommentar:",
+                "Frokostforedrag kommentar:",
+                "Alternativt arrangement kommentar:",
+                "Start-up kommentar:",
+                "Bedrift-til-bedrift kommentar:",
+                "Lunsjpresentasjon kommentar:",
+                "Bedriftspresentasjon kommentar:",
+                "BedEx kommentar:",
+            ]
+        )
+        for ci in companyInterests:
+            print(ci)
```

Comment:
Could be nice for the readability of it all to name it 'companyInterest' and not ci, and remove the print statement